src/Windows/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -24,15 +26,28 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
 
         private RFormatOptions _formatOptions = new RFormatOptions();
 
-        public REditorSettings(ICoreShell coreShell): base(coreShell, RContentTypeDefinition.ContentType) { }
-        public REditorSettings(IEditorSettingsStorage storage) : base(storage) { }
+        public REditorSettings(ICoreShell coreShell) : base(coreShell, RContentTypeDefinition.ContentType) {
+            CreateLintOptions();
+        }
+
+        public REditorSettings(IEditorSettingsStorage storage) : base(storage) {
+            CreateLintOptions();
+        }
 
         public override void ResetSettings() {
             _formatOptions = new RFormatOptions();
-            LintOptions = new LintOptions();
+            CreateLintOptions();
             base.ResetSettings();
         }
 
+        private void CreateLintOptions() {
+            LintOptions = new LintOptions();
+            LintOptions.PropertyChanged += _OnLintOptionsPropertyChanged;
```

Comment:
There are three...

`ISettingsStorage` (which is actually writable) is implemented over roaming store (`SVsSettingsPersistenceManager`). This is what we *partially* use for `IRSettings`. Mostly for settings we store manually (without dialog page) such as connections, most recently used settings in SQL publishing etc.

There is `IEditorSettingsStorage` which is implemented by `LanguageSettingsStorage` over `SVsTextManager` and `LANGPREFERENCES3`

DialogPage settings are loaded by package by calling `GetAutomationObject` on every option page we have. This loads settings from virtual registry key specified in the dialog page and sets appropriate fields in IRSettings.


It is actually `IWritableEditorSettingsStorage`